app/components/koi/tables/header_cell_component.rb

```diff
@@ -3,6 +3,39 @@
 module Koi
   module Tables
     class HeaderCellComponent < Katalyst::Tables::HeaderCellComponent
+      using Katalyst::HtmlAttributes::HasHtmlAttributes
+
+      attr_reader :width, :type
+
+      def initialize(table, attribute, label: nil, link: {}, width: nil, type: :text, **html_attributes)
+        @width = width
+        @type = type
+
+        super(table, attribute, label:, link:, **html_attributes)
+      end
+
+      def default_html_attributes
+        super.merge_html(class: [type_class, width_class].compact.join(" "))
+      end
+
+      private
+
+      def width_class
+        return if width.blank?
+
+        case width
+        when "medium"
+          "katalyst--tables-column-m"
+        when "large"
+          "katalyst--tables-column-l"
+        else
+          "katalyst--tables-column-s"
+        end
+      end
+
+      def type_class
+        "katalyst--tables-column-#{type.to_s.dasherize}"
```

Comment:
I'm 100% against this practice – generating class names in this way makes it virtually impossible to find where the class is used.

AFAICT the only advantage to this approach is that it avoids having to pre-define header components that declare their type. Why is that valuable? Seems like saving a few keystrokes now for a lot less clarity for future readers. As library authors we should be optimising for readability.